src/frequenz/dispatch/_managing_actor.py

```diff
@@ -0,0 +1,178 @@
+
+        status_receiver = dispatcher.running_status_change.new_receiver()
+
+        dispatch_runner = DispatchManagingActor(
+            actor=my_actor,
+            dispatch_type="EXAMPLE",
+            running_status_receiver=status_receiver,
+            updates_sender=dispatch_updates_channel.new_sender(),
+        )
+
+        await asyncio.gather(dispatcher.start(), dispatch_runner.start())
+    ```
+    """
+
+    def __init__(
+        self,
+        actor: Actor,
+        dispatch_type: str,
+        running_status_receiver: Receiver[Dispatch],
+        updates_sender: Sender[DispatchUpdate] | None = None,
+    ) -> None:
+        """Initialize the dispatch handler.
+
+        Args:
+            actor: The actor to manage.
+            dispatch_type: The type of dispatches to handle.
+            running_status_receiver: The receiver for dispatch running status changes.
+            updates_sender: The sender for dispatch events
+        """
+        super().__init__()
+        self._dispatch_rx = running_status_receiver
+        self._actor = actor
+        self._dispatch_type = dispatch_type
+        self._updates_sender = updates_sender
+
+    def _start_actor(self) -> None:
+        """Start the actor."""
+        if self._actor.is_running:
+            _logger.warning("Actor %s is already running", self._actor.name)
+        else:
+            self._actor.start()
+
+    async def _stop_actor(self, msg: str) -> None:
+        """Stop the actor.
+
+        Args:
+            msg: The message to be passed to the actor being stopped.
+        """
+        if self._actor.is_running:
+            await self._actor.stop(msg)
+        else:
+            _logger.warning("Actor %s is not running", self._actor.name)
+
+    async def _run(self) -> None:
+        """Wait for dispatches and handle them."""
+        async for dispatch in self._dispatch_rx:
+            await self._handle_dispatch(dispatch=dispatch)
+
+    async def _handle_dispatch(self, dispatch: Dispatch) -> None:
+        """Handle a dispatch.
+
+        Args:
+            dispatch: The dispatch to handle.
+        """
+        running = dispatch.running(self._dispatch_type)
+        match running:
+            case RunningState.STOPPED:
+                _logger.info("Stopped by dispatch %s", dispatch.id)
+                await self._stop_actor("Dispatch stopped")
+            case RunningState.RUNNING:
+                if self._updates_sender is not None:
+                    _logger.info("Updated by dispatch %s", dispatch.id)
+                    await self._updates_sender.send(
+                        DispatchUpdate(
+                            components=dispatch.selector,
+                            dry_run=dispatch.dry_run,
+                            options=dispatch.payload,
+                        )
+                    )
+
+                _logger.info("Started by dispatch %s", dispatch.id)
+                self._start_actor()
```

Comment:
I just realized this is tricky. The actor needs to make sure it reads from the updates channel in the `_run()` method before doing any other work, otherwise it might be running with a missing configuration until the update message arrives. I think we should put this more clearly in the docs, including the example. But we can do it as a followup PR.

Reply:
* https://github.com/frequenz-floss/frequenz-dispatch-python/issues/56